reporter.go

```diff
@@ -520,88 +476,85 @@ func (r *Reporter) Compare(out io.Writer) {
-	}
-
-	awst, err := ptypes.Timestamp(r.after.StartWalk)
+// printWalkSummary prints some information about the given walk.
+func (r *Reporter) printWalkSummary(out io.Writer, walk *fspb.Walk) {
+	awst, err := ptypes.Timestamp(walk.StartWalk)
 	if err != nil {
 		log.Fatalf("unable to convert after walk start timestamp: %v", err)
 	}
-	awet, err := ptypes.Timestamp(r.after.StopWalk)
+	awet, err := ptypes.Timestamp(walk.StopWalk)
 	if err != nil {
 		log.Fatalf("unable to convert after walk stop timestamp: %v", err)
 	}
-	fmt.Fprintln(out, "Walk (After)")
-	fmt.Fprintf(out, "  - ID: %s\n", r.after.Id)
+	fmt.Fprintf(out, "  - ID: %s\n", walk.Id)
 	fmt.Fprintf(out, "  - Start Time: %s\n", awst)
 	fmt.Fprintf(out, "  - Stop Time: %s\n", awet)
+}
+
+// PrintReportSummary prints a few key information pieces around the Report.
+func (r *Reporter) PrintReportSummary(out io.Writer, report *Report) {
+	fmt.Fprintln(out, "===============================================================================")
+	fmt.Fprintln(out, "Report Summary:")
+	fmt.Fprintln(out, "===============================================================================")
+	fmt.Fprintf(out, "Host name: %s\n", report.WalkAfter.Hostname)
+	fmt.Fprintf(out, "Report config used: %s\n", r.configPath)
+	if report.WalkBefore != nil {
+		fmt.Fprintln(out, "Walk (Before)")
+		r.printWalkSummary(out, report.WalkBefore)
+	}
+	fmt.Fprintln(out, "Walk (After)")
+	r.printWalkSummary(out, report.WalkAfter)
 	fmt.Fprintln(out)
 }
 
 // PrintRuleSummary prints the configs and policies involved in creating the Walk and Report.
-func (r *Reporter) PrintRuleSummary(out io.Writer) {
+func (r *Reporter) PrintRuleSummary(out io.Writer, report *Report) {
 	fmt.Fprintln(out, "===============================================================================")
 	fmt.Fprintln(out, "Rule Summary:")
 	fmt.Fprintln(out, "===============================================================================")
-	fmt.Fprintln(out)
 	fmt.Fprintln(out, "Client Policy:")
-	if r.before == nil {
-		fmt.Fprintln(out, proto.MarshalTextString(r.after.Policy))
+	if report.WalkBefore == nil {
+		fmt.Fprintln(out, proto.MarshalTextString(report.WalkAfter.Policy))
 	} else {
-		diff := cmp.Diff(r.before.Policy, r.after.Policy)
+		diff := cmp.Diff(report.WalkBefore.Policy, report.WalkAfter.Policy)
 		if diff != "" {
 			fmt.Fprintln(out, "Diff:")
 			fmt.Fprintln(out, diff)
 			fmt.Fprintln(out, "Before:")
 		}
-		fmt.Fprintln(out, proto.MarshalTextString(r.before.Policy))
+		fmt.Fprintln(out, proto.MarshalTextString(report.WalkBefore.Policy))
 	}
 	fmt.Fprintln(out, "Report Config:")
 	fmt.Fprintln(out, proto.MarshalTextString(r.config))
 }
 
 // UpdateReviewProto updates the reviews file to the reviewed version to be "last known good".
-func (r *Reporter) UpdateReviewProto(ctx context.Context) error {
+func (r *Reporter) UpdateReviewProto(ctx context.Context, walkFile *WalkFile, reviewFile string) error {
```

Comment:
And same here as with `PrintDiffSummary`: seems like receiver `r` is now unused and `UpdateReviewProto` could become a simple function, as opposed to the r's method.